File: cyclic_hanoi_mimic.cpp
#include<iostream>

using namespace std;

long long int ccw(long long int x2, long long int * x19, long long int * x20, 
		long long int * x21);
long long int cw(long long int x2, long long int * x19, long long int * x20, 
		long long int * x21);
int main(){

	long long intint x2 = 0;
	long long int * x19 = new long long int[16];
	long long int * x20 = new long long int[16];
	long long int * x21 = new long long int[16];
	

	int x15 = 255;
	*x19 = x15;
	*x20 = x15;
	*x21 = x15;
	
	int x4 = 10;
	x15 = x4;

	loop:
		x19++;
		*x19 = x15;
		x15 -= 1;
		if (x15 != 0){
			goto loop;
		}
		else{
			goto chanoi;
		}

	chanoi:
		if (*x19 == 255){
			goto stop;
		}
		x2 = 2*ccw(x2) + 1;
	stop:
	
		delete[] x19;
		delete[] x20;
		delete[] x21;		

}

long long int ccw

File: cyclic_hanoi_mimic.py
print("Starting calculating number of steps for Cyclic Hanoi")


x2 = 0


x19 = []
x20 = []
x21 = []











def cw(n):
	if n==0:
		return 0
	if n==1:
		return 1
	else:
		return 2*ccw(n-1) + 1

def ccw(n):
	if n==0:
		return 0
	if n==1:
		return 2
	else:
		return 2*ccw(n-1) + cw(n-1) + 2

def th(n):
	print(2*ccw(n-1) + 1)


th(5)


